pkg/reconciler/openshift/namespace/namespace.go

```diff
@@ -0,0 +1,278 @@
+	"fmt"
+	"github.com/tektoncd/operator/pkg/client/listers/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/reconciler/openshift"
+	"github.com/tektoncd/operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"strings"
+
+	"github.com/markbates/inflect"
+	"go.uber.org/zap"
+	admissionv1 "k8s.io/api/admission/v1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	admissionlisters "k8s.io/client-go/listers/admissionregistration/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/kmp"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/ptr"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/system"
+	"knative.dev/pkg/webhook"
+	certresources "knative.dev/pkg/webhook/certificates/resources"
+)
+
+// reconciler implements the AdmissionController for resources
+type reconciler struct {
+	webhook.StatelessAdmissionImpl
+	pkgreconciler.LeaderAwareFuncs
+
+	key  types.NamespacedName
+	path string
+
+	withContext func(context.Context) context.Context
+
+	client             kubernetes.Interface
+	mwhlister          admissionlisters.MutatingWebhookConfigurationLister
+	secretlister       corelisters.SecretLister
+	tektonConfigLister v1alpha1.TektonConfigLister
+
+	disallowUnknownFields bool
+	secretName            string
+}
+
+var _ controller.Reconciler = (*reconciler)(nil)
+var _ pkgreconciler.LeaderAware = (*reconciler)(nil)
+var _ webhook.AdmissionController = (*reconciler)(nil)
+var _ webhook.StatelessAdmissionController = (*reconciler)(nil)
+
+// Reconcile implements controller.Reconciler
+func (ac *reconciler) Reconcile(ctx context.Context, key string) error {
+	logger := logging.FromContext(ctx)
+
+	if !ac.IsLeaderFor(ac.key) {
+		logger.Debugf("Skipping key %q, not the leader.", ac.key)
+		return nil
+	}
+
+	// Look up the webhook secret, and fetch the CA cert bundle.
+	secret, err := ac.secretlister.Secrets(system.Namespace()).Get(ac.secretName)
+	if err != nil {
+		logger.Errorw("Error fetching secret", zap.Error(err))
+		return err
+	}
+	caCert, ok := secret.Data[certresources.CACert]
+	if !ok {
+		return fmt.Errorf("secret %q is missing %q key", ac.secretName, certresources.CACert)
+	}
+
+	// Reconcile the webhook configuration.
+	return ac.reconcileMutatingWebhook(ctx, caCert)
+}
+
+// Path implements AdmissionController
+func (ac *reconciler) Path() string {
+	return ac.path
+}
+
+// Admit implements AdmissionController
+func (ac *reconciler) Admit(ctx context.Context, request *admissionv1.AdmissionRequest) *admissionv1.AdmissionResponse {
+	if ac.withContext != nil {
+		ctx = ac.withContext(ctx)
+	}
+
+	logger := logging.FromContext(ctx)
+
+	// Need to handle both, create and update operations for a namespace
+	switch request.Operation {
+	case admissionv1.Create:
+	case admissionv1.Update:
+	default:
```

Comment:
If it need to handle `Create` and `Update`, it probably needs a `fallthrough` doesn't it ?

Reply:
i think the request will fail and go back, nothing will happen on etcd

Reply:
but something to test and check

Reply:
there are 2 more operator, CONNECT and DELETE, so this is fine